geth/account/accounts.go

```diff
@@ -14,30 +13,32 @@ import (
 	gethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/status-im/status-go/extkeys"
-	"github.com/status-im/status-go/geth/common"
-	"github.com/status-im/status-go/geth/rpc"
 )
 
 // errors
 var (
-	ErrAddressToAccountMappingFailure  = errors.New("cannot retrieve a valid account for a given address")
-	ErrAccountToKeyMappingFailure      = errors.New("cannot retrieve a valid key for a given account")
-	ErrWhisperIdentityInjectionFailure = errors.New("failed to inject identity into Whisper")
-	ErrWhisperClearIdentitiesFailure   = errors.New("failed to clear whisper identities")
-	ErrNoAccountSelected               = errors.New("no account has been selected, please login")
-	ErrInvalidMasterKeyCreated         = errors.New("can not create master extended key")
+	ErrAddressToAccountMappingFailure = errors.New("cannot retrieve a valid account for a given address")
+	ErrAccountToKeyMappingFailure     = errors.New("cannot retrieve a valid key for a given account")
+	ErrNoAccountSelected              = errors.New("no account has been selected, please login")
+	ErrInvalidMasterKeyCreated        = errors.New("can not create master extended key")
 )
 
-// Manager represents account manager interface
+// GethServiceProvider provides required geth services.
+type GethServiceProvider interface {
```

Comment:
it is not services, just a couple of internal geth objects. is it possible to pass account manager and keystore without provider after refactoring that we talked about?

Reply:
About naming does `GethProvider` could make more sense here?

Changes related on how StatusBackend manages those geth objects have been moved out this pull request as there were many changes for a single pull request. 

Anyway, i'm not sure about directly receiving manager and keystore objects. If we do, `StatusBackend` will need extra logic to manage updates on those objects whenever the underlying node is restarted. 
On the other hand if we just pass an object accomplishing this interface (it doesn't need to be NodeManager), this logic will no longer be needed, as both `AccountManager` and `AccountKeyStore` will return updated objects, and the less logic the better.

Reply:
We can also split it into to separate interfaces if figuring out a common name is hard: `AccountManagerProvider` and `KeyStoreProvider`. 

Regarding passing these objects as params, we talked about it and we would like to avoid it in this PR. It requires additional logic in `StatusBackend`. During node restart, the managers are kept as the same instances but geth `Node` is created again and both `accounts.Manager` and `keystore.KeyStore` are changed.

Reply:
> Anyway, i'm not sure about directly receiving manager and keystore objects. If we do, StatusBackend will need extra logic to manage updates on those objects whenever the underlying node is restarted.

you can create new objects every time when backend calls start, the only thing that needs to be persisted is selected account (also i am not totally sure why, i would check if react part actually relies on it in any way)

> On the other hand if we just pass an object accomplishing this interface (it doesn't need to be NodeManager)

this is definitely not a good pattern, it leads to weird tests where you need to **MOCK** objects that should be just provided as arguments and constant checking if node is running when such object returns.

Reply:
> you can create new objects every time when backend calls start, the only thing that needs to be persisted is selected account (also i am not totally sure why, i would check if react part actually relies on it in any way)

That's one thing and another thing is that `account.Manager` is passed to `transactions.Manager`. So we need to replace `transactions.Manager` as well. I am not saying it's not gonna work, I am saying that we can do it step by step. This refactoring will make another PR with your suggestions trivial, provided there won't be any other surprises.

> this is definitely not a good pattern, it leads to weird tests where you need to MOCK

You don't need to pass a mock. It gives a choice to pass whatever object that matches the interface. It can be a `NodeManager` or it can be a new object that provides real `KeyStore` and geth `accounts.Manager` objects.

Reply:
> It can be a NodeManager 

that's an overkill for account manager unit tests, you also have to start node, because account manager checks for node to be running

> or it can be a new object that provides real KeyStore and geth accounts.Manager objects.

thats a mock basically, no?

Reply:
Right, so no one should really use NodeManager even though it matches the interface.

Hm, technically yes :D I meant that it doesn't need to be a generated mock struct that we use in other unit tests. It can be a very basic struct with two methods. It would not be very different from providing the same object as arguments.

Reply:
@dshulyak I agree with your point, and we shouldn't stop on that refactoring.
But I have one concern.
It looks like that if we go your way now, we end up with much bigger PR and this change, which is a step forward, will be delayed.
I'd rather have a step forward sooner than a perfect solution someday. Who knows, maybe some more urgent and exciting tasks will appear, and we won't merge anything to our codebase then.